docs/website/contents/for-developers/HardForkCombinatorRollbackMonotonicityProposal.md

```diff
@@ -0,0 +1,285 @@
+
+That is a complete answer.
+In particular, there is nothing else this simplified node needs to predict.
+
+If we assume Chain Growth in addition to Common Prefix, then we can further refine the answer.
+
+- Chain Growth ensures that every window of scg (ie "the s parameter of the papers' CG property", aka "one stability window") slots will contain at least k blocks on C.
+- Because of the Header-Body Split, L must contain enough information for the node to correctly judge the validity of any header that is no more than 3k/f after the first block after L on C.
+
+Note that, if necessary, L can be ticked in order to validate the first header after L on C, without requiring any predictions.
+In the absence of the corresponding block body---ie because of the Header-Body Split---the assessment of subsequent headers on C will require predictions.
+Thus a prediction range of at least 3k/f ensures at least k blocks after the first header, and hence k+1 blocks in total, as necessary.
+
+### Question 2
+
+What behaviors could arise only if Chain Growth were violated?
+
+- For example, suppose R=k and the k+1th block after L on C was more than 3k/f slots after the first block after L on C.
+- In this case, the node might not be able to switch from D to C despite R <= k.
+  It can't switch to C without downloading the R+1 blocks, which it can't do without validating the R+1 headers, which it can't do without predicting more than 3k/f slots after the first header, which it might not be able to do.
+
+For example, suppose that L is the result of a block in relative slot number 7k/f - 2 within some epoch (ie "slightly more than scg before the end of the epoch").
+Because there might be a block in relative slot 7k/f - 1---which would change the nonce used for the next epoch---L is unable to validate headers in the next epoch.
+If the first header after L on C is indeed in slot 7k/f - 1, then Chain Growth ensures the k+1th header after L on C will still be in this same epoch, and so L can validate it.
+If the first header after L on C is instead after slot 7k/f - 1, then ticking L to that first header will tick it past slot 7k/f - 1, and so determines the next epoch's nonce.
+That's likely sufficient for the next k headers, unless there is an extremely severe Chain Growth violation.
+(In the non-simplified node, the prediction is always restricted to exactly 3k/f slots, despite the data in question actually advancing an entire epoch at a time.
+Thus regardless of where the first header is, any Chain Growth violation in the 3k/f slots after it would prevent switch to C when R=k.)
+
+That is a complete answer, but there are a few particulars worth emphasizing.
+
+- No other behavior of the simplified node will change due to a Chain Growth violation.
+  (This is actually false, but only because we use predictions in the leadership check.
+  One option to make this true is to instead use ticking in the leadership check.
```

Comment:
Why ticking would make this true?

Reply:
Because you can only forecast so far, but you can tick indefinitely (in theory, at least). EG if there's a block in the slot 0 of an epoch, but there no block in the next 3k/f slots, then today's node wouldn't be able to lead even if it were elected in the next slot (ie slot 3k/f+1 of the epoch), since it couldn't forecast far enough to see that. On the other hand, it can tick to that slot.

(In theory ticking has no limit, but I think the Cardano ledger _currently_ assumes you can't tick across more than one epoch boundary.)